fearless/Common/Services/PricesService.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+import SSFModels
+import RobinHood
+
+protocol PricesServiceProtocol {
+    func startPricesObserving(for chainAssets: [ChainAsset], currencies: [Currency])
+    func updatePrices()
+}
+
+final class PricesService: PricesServiceProtocol {
+    static let shared: PricesServiceProtocol = PricesService.create()
+    private let priceLocalSubscriber = PriceLocalStorageSubscriberImpl.shared
+    private let assetRepository: AnyDataProviderRepository<AssetModel>
+    private let operationQueue: OperationQueue
+    private let logger: Logger
+    private var pricesProvider: AnySingleValueProvider<[PriceData]>?
+    private var assets: [AssetModel] = []
+
+    private init(
+        assetRepository: AnyDataProviderRepository<AssetModel>,
+        operationQueue: OperationQueue,
+        logger: Logger
+    ) {
+        self.assetRepository = assetRepository
+        self.operationQueue = operationQueue
+        self.logger = logger
+    }
+
+    func startPricesObserving(for chainAssets: [SSFModels.ChainAsset], currencies: [SSFModels.Currency]) {
+        pricesProvider = priceLocalSubscriber.subscribeToPrices(
+            for: chainAssets,
+            currencies: currencies,
+            listener: self
+        )
+        self.assets = chainAssets.map { $0.asset }.uniq(predicate: { assetModel in
+            assetModel.id
+        })
+    }
+
+    func updatePrices() {
+        pricesProvider?.refresh()
+    }
+}
+
+extension PricesService: PriceLocalSubscriptionHandler {
+    func handlePrice(
+        result _: Result<PriceData?, Error>,
+        chainAsset _: ChainAsset
+    ) {}
+
+    func handlePrices(result: Result<[PriceData], Error>) {
```

Comment:
is it possible to change this delegate method by changing [PriceData] to [ChainAsset: PriceData] ?